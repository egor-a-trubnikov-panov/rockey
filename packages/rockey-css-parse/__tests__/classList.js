import createParser from '../lib/parse';

const parse = createParser({
  getClassName: component => `c-${component}-hash`,
  getMixinClassName: component => `m-${component}-hash`,
  getAnimationName: component => `a-${component}-hash`,
});

test('classList #1', () => {
  const parsed = parse(`
    Button {
      color: red;
    }
  `);

  expect(parsed.classList).toMatchSnapshot();
});

test('classList #2', () => {
  const parsed = parse(`
    Button, PrimaryButton, SuccessButton, span.hey {
      color: red;
    }
  `);

  expect(parsed.classList).toMatchSnapshot();
});
