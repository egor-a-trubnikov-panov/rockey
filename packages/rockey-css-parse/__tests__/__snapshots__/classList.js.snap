// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classList #1 1`] = `
Object {
  "Button": "c-Button-hash",
}
`;

exports[`classList #2 1`] = `
Object {
  "Button": "c-Button-hash",
  "PrimaryButton": "c-PrimaryButton-hash",
  "SuccessButton": "c-SuccessButton-hash",
  "span": "span.hey",
}
`;
